src/nvim/regexp.c

```diff
@@ -6984,10 +6984,14 @@ regprog_T *vim_regcomp(char_u *expr_arg, int re_flags)
   /*
    * First try the NFA engine, unless backtracking was requested.
    */
-  if (regexp_engine != BACKTRACKING_ENGINE)
-    prog = nfa_regengine.regcomp(expr, re_flags);
-  else
+  if (regexp_engine != BACKTRACKING_ENGINE) {
+    prog = nfa_regengine.regcomp(expr,
+                                 re_flags + (regexp_engine == AUTOMATIC_ENGINE
+                                             ? RE_AUTO
+                                             : 0));
```

Comment:
Golly I'm so not a fan of vertical alignment. Makes the code look all over the place. Any tips on how I should layout this?